dpnp/dpnp_iface_searching.py

```diff
@@ -82,46 +96,90 @@ def argmax(x1, axis=None, out=None):
     --------
     >>> import dpnp as np
     >>> a = np.arange(6).reshape((2, 3)) + 10
-    >>> a.shape
-    (2, 3)
-    >>> [i for i in a]
-    [10, 11, 12, 13, 14, 15]
+    >>> a
+    array([[10, 11, 12],
+           [13, 14, 15]])
     >>> np.argmax(a)
-    5
+    array(5)
+
+    >>> np.argmax(a, axis=0)
+    array([1, 1, 1])
+    >>> np.argmax(a, axis=1)
+    array([2, 2])
+
+    >>> b = np.arange(6)
+    >>> b[1] = 5
+    >>> b
+    array([0, 5, 2, 3, 4, 5])
+    >>> np.argmax(b)  # Only the first occurrence is returned.
+    array(1)
+
+    >>> x = np.arange(24).reshape((2, 3, 4))
+    >>> res = np.argmax(x, axis=1, keepdims=True) # Setting keepdims to True
+    >>> res.shape
+    (2, 1, 4)
 
     """
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc:
-        if axis is not None:
-            pass
-        elif out is not None:
-            pass
-        else:
-            result_obj = dpnp_argmax(x1_desc).get_pyobj()
-            result = dpnp.convert_single_elem_array_to_scalar(result_obj)
+    dpt_array = dpnp.get_usm_ndarray(a)
+    result = dpnp_array._create_from_usm_ndarray(
+        dpt.argmax(dpt_array, axis=axis, keepdims=keepdims)
+    )
+
+    if out is None:
+        return result
+    else:
+        if out.shape != result.shape:
+            raise ValueError(
+                f"Output array of shape {result.shape} is needed, got {out.shape}."
+            )
+        elif not isinstance(out, dpnp_array):
+            if isinstance(out, dpt.usm_ndarray):
+                out = dpnp_array._create_from_usm_ndarray(out)
+            else:
+                raise TypeError(
+                    "Output array must be any of supported type, but got {}".format(
+                        type(out)
+                    )
+                )
 
-            return result
+        dpnp.copyto(out, result, casting="safe")
 
-    return call_origin(numpy.argmax, x1, axis, out)
+        return out
```

Comment:
It's exactly the same for `argmin`, `argmax`, `min` and `max` function. Can we separate that to some helper function?